setup.py

```diff
@@ -59,6 +59,9 @@
         'BeautifulSoup4',  # VERY weak requirement, still used in one of the tests
         'httpretty>=0.9.4',  # Introduced py 3.6 support
         'nose>=1.3.4',
```

Comment:
shouldn't `nose` be removed ?  (I left a comment related to this above https://github.com/datalad/datalad/pull/6273/files#diff-b73bd279c22df79a816676fed59fb182bfdca0cf5ae9d0d26d3a71f399ddd94aR50 )

Reply:
Only a couple test files were converted from nose to pytest.  There are others that still import `nose`.

Reply:
but we aren't running those tests in this PR right?  I think we should strip away `nose` from requirements, develop in env without nose locally, and keep expanding list of test files while converting them to pytest.  Or would you suggest some other strategy?

Reply:
There are still some matrix entries that run other tests in DataLad, hence the need to keep installing `nose`.